core/src/main/java/org/gbif/pipelines/core/functions/transforms/ExtendedRecordToLocationTransformer.java

```diff
@@ -0,0 +1,187 @@
+package org.gbif.pipelines.core.functions.transforms;
+
+import org.gbif.api.vocabulary.Continent;
+import org.gbif.api.vocabulary.Country;
+import org.gbif.common.parsers.ContinentParser;
+import org.gbif.common.parsers.CountryParser;
+import org.gbif.common.parsers.core.ParseResult;
+import org.gbif.dwca.avro.Location;
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueLineageRecord;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueType;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+import org.gbif.pipelines.core.functions.interpretation.error.LineageType;
+import org.gbif.pipelines.io.avro.ExtendedRecord;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.TupleTag;
+
+/**
+ * This function converts an extended record to an interpreted KeyValue of occurenceId and Event.
+ * This function returns multiple outputs,
+ * a. Interpreted version of raw temporal data as KV<String,Event>
+ * b. Issues and lineages applied on raw data to get the interpreted result, as KV<String,IssueLineageRecord>
+ */
+public class ExtendedRecordToLocationTransformer extends DoFn<ExtendedRecord, KV<String, Location>> {
+
+  /**
+   * tags for locating different type of outputs send by this function
+   */
+  public static final TupleTag<KV<String, Location>> locationDataTag = new TupleTag<KV<String, Location>>();
+  public static final TupleTag<KV<String, IssueLineageRecord>> locationIssueTag =
+    new TupleTag<KV<String, IssueLineageRecord>>();
+
+  @ProcessElement
+  public void processElement(ProcessContext ctx) {
+    ExtendedRecord record = ctx.element();
+    Location loc = new Location();
+
+    Map<CharSequence, List<Issue>> fieldIssueMap = new HashMap<CharSequence, List<Issue>>();
+    Map<CharSequence, List<Lineage>> fieldLineageMap = new HashMap<CharSequence, List<Lineage>>();
+    //mapping raw record with interpreted ones
+    loc.setOccurrenceID(record.getId());
+    loc.setLocationID(record.getCoreTerms().get(DwCATermIdentifier.locationID.getIdentifier()));
+    loc.setHigherGeographyID(record.getCoreTerms().get(DwCATermIdentifier.higherGeographyID.getIdentifier()));
+    loc.setHigherGeography(record.getCoreTerms().get(DwCATermIdentifier.higherGeography.getIdentifier()));
+
+    /*
+      Interpreting Continent
+     */
+
+    CharSequence rawContinent = record.getCoreTerms().get(DwCATermIdentifier.continent.getIdentifier());
+    String interpretedContinent = null;
+    if (rawContinent != null) {
+      List<Issue> issues = new ArrayList<>();
```

Comment:
change these variables to be Collection.singletonList, this makes clear that is a list of 1 element